packages/core/src/internal/proc.js

```diff
@@ -483,7 +475,7 @@ export default function proc(
         return
       }
       if (isEnd(input) && !maybe) {
-        cb(CHANNEL_END)
+        cb({ ...input, [TERMINATE]: true })
```

Comment:
can u think of any use case for extending the `input` with a flag? maybe we could just pass special completion value - `TERMINATE` here, like this:
```js
if (isEnd(input) && !maybe) {
  cb(TERMINATE)
  return
}
```

I know I'm probably one who wrote this spread here, but can't think of any use case of it now.

If we would follow that path, then we could skip implementing `isTerminated` altogether and just use direct `TERMINATE` comparisong, i.e.
```js
 } else if (arg === TERMINATE) {
```
and
```js
const isCompleted = res => res && (isEnd(res) || res === TERMINATE || res === TASK_CANCEL)
```

Reply:
checked, also didn't find any cases, updated